### PR TITLE
Add vector symbolic support

### PR DESCRIPTION
**Pull request by YingboMa**
Needs https://github.com/JuliaSymbolics/Symbolics.jl/pull/353 and https://github.com/SciML/ModelingToolkit.jl/pull/1222

### CODE REVIEW

File: test/downstream/symbol_indexing.jl [hunk 1]
```diff
@@ -152,3 +152,18 @@ plot(sol,vars=(lorenz2.x,α))
...
+ps = @parameters p[1:3] = [1, 2, 3]
+D = Differential(t)
+eqs = [
+       collect(D.(x) .~ x)
+       D(y) ~ norm(x)*y - x[1]
+      ]
+@named sys = ODESystem(eqs, t, [sts...;], [ps...;])
+prob = ODEProblem(sys, [], (0, 1.0))
+sol = solve(prob, Tsit5())
+sol[x] isa Vector{<:Vector}
```
Comment (ChrisRackauckas) [inline test/downstream/symbol_indexing.jl:168]: Test `sol[2x]` as well?

Comment (YingboMa) [reply test/downstream/symbol_indexing.jl:168]: I don't think that ever worked.

Comment (ChrisRackauckas) [reply test/downstream/symbol_indexing.jl:168]: it does for scalars IIRC

Comment (YingboMa) [reply test/downstream/symbol_indexing.jl:168]: I am pretty sure it doesn't.
```julia
julia> sol[y]
10-element Vector{Float64}:
   1.0
   1.2229920909829024
   1.7300850610048448
   2.739731116321167
   5.169184888556082
  11.50405236824545
  30.6464424321771
  96.83020784927531
 361.2079995651449
 454.1481546480951

julia> sol[2y]
ERROR: Indexing symbol 2y(t) is unknown.
Stacktrace:
  [1] error(s::String)
    @ Base ./error.jl:33
  [2] DEFAULT_OBSERVED(sym::Num, u::Vector{Float64}, p::Vector{Float64}, t::Float64)
    @ SciMLBase ~/src/julia/SciMLBase/src/scimlfunctions.jl:4
  [3] _broadcast_getindex_evalf
    @ ./broadcast.jl:648 [inlined]
  [4] _broadcast_getindex
    @ ./broadcast.jl:621 [inlined]
  [5] getindex
    @ ./broadcast.jl:575 [inlined]
  [6] copy
    @ ./broadcast.jl:922 [inlined]
  [7] materialize
    @ ./broadcast.jl:883 [inlined]
  [8] observed(A::ODESolution{Float64,, sym::Num, i::Colon)
    @ SciMLBase ~/src/julia/SciMLBase/src/solutions/solution_interface.jl:99
  [9] getindex(A::ODESolution{Float64,, sym::Num)
    @ SciMLBase ~/src/julia/SciMLBase/src/solutions/solution_interface.jl:60
 [10] top-level scope
    @ REPL[125]:1
```

Comment (ChrisRackauckas) [reply test/downstream/symbol_indexing.jl:168]: didn't you do that PR?

**Comment by ChrisRackauckas** [comment]
@YingboMa did you PR the updated version? Don't want to lose that again 😅

**Pull request merged**